tests/test_util.py

```diff
@@ -0,0 +1,180 @@
+import asyncio
```

Comment:
Some general comments on this whole file ...

This is a really cool system you've built but I'd like to see some documentation on it.  A little header block explaining how this is expected to be run, any notes about how to build new tests, would be great.

Additionally, Pythonic doc-blocks for the functions would be good.  I know its a lot to ask for something that's done on your free time, but I just spent a year of my life documenting code so asking for a little bit of docs from others seems fair :D 

Pythonic docblocks look something like:
```
def myfunction(arg1, arg2, arg3):
    """This is a brief description of the code

    Args:
        arg1: str that does this
        arg2: int that does that
        arg3: dict that does whatever and has these elements

    Returns:
        A whole sandwich
    """
```

'self' doesn't need docs when passed as a parameter because that's a language feature.  If you want to do python3 type hinting for arguments that is also great but not required.

Thanks for cooperation :)